### PR TITLE
Fix flip array for groups in select in segments filter

### PR DESCRIPTION
**Pull request by kuzmany**
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | staging for features or enhancements / 3.0 for bug fixes 
| Bug fix?                               | yes/no
| New feature?                           | yes/no
| Deprecations?                          | yes/no
| BC breaks?                             | yes/no
| Automated tests included?              | yes/no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Issue coming from our plugin what add new segment filters. Probably code review should enough and  check If this change not ruined any select/multiselect/country field in segment filter

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/9339"></a>

### CODE REVIEW

**Comment by RCheesley** [comment]
@kuzmany I don't see any link to an issue or explanation/test instructions for this PR - would you be able to add some information please? Thank you!

**Comment by dadarya0** [comment]
@kuzmany could you please add description and step to test this PR.

**Comment by kuzmany** [comment]
@dadarya0 I added description

> Issue coming from our plugin what add new segment filters. Probably code review should enough and  check If this change not ruined any select/multiselect/country field in segment filter

**Comment by RCheesley** [comment]
Bumping from the release due to lack of testing/code review.

**Comment by RCheesley** [comment]
@kuzmany this will need to be rebased to 4.1 please

**Comment by kuzmany** [comment]
@RCheesley done

File: app/bundles/CoreBundle/Helper/ArrayHelper.php [hunk 1]
```diff
@@ -94,6 +94,27 @@ function ($value) {
...
+        if (!is_array(end($masterArrays))) {
+            return array_flip($masterArrays);
+        }
+
+        return array_map(
+            function (array $subArray) {
+                return array_flip($subArray);
+            },
+            $masterArrays
+        );
```
Comment (escopecz) [inline app/bundles/CoreBundle/Helper/ArrayHelper.php]: We can use arrow functions now 🎉 
```suggestion
        return array_map(
            fn (array $subArray) => array_flip($subArray),
            $masterArrays
        );
```

**Review by escopecz — DISMISSED** [review]
The code looks good. I have one suggestion to use shorter syntax but either way it's good by me.

**Comment by RCheesley** [comment]
Seems to be flagging a reduction in code coverage here - is there anything we can do to improve that @escopecz @kuzmany ?

**Comment by RCheesley** [comment]
Based on @escopecz review and talking through we don't need to increase the test coverage here because it's flagging due to the removal of code, so marking this as RTC.

**Pull request merged**